lib/internal/readline/interface.js

```diff
@@ -449,8 +454,15 @@ class Interface extends InterfaceConstructor {
 
     // If the trimmed line is empty then return the line
     if (StringPrototypeTrim(this.line).length === 0) return this.line;
-
-    if (this.history.length === 0 || this.history[0] !== this.line) {
+    const normalizedLine = StringPrototypeReplaceAll(this.line, '\n', '\r');
```

Comment:
I understand that this works fine, I just fear it might make it more difficult to adjust the methods further. As developer, I would be surprised about the behavior change for \r and \n. Is there a possibility that we abstract the conversion somewhere in a central point that we know it will only ever happen in exactly two places and changing other code would not need to know about it?